controllers/machinedeletionremediation_controller.go

```diff
@@ -167,13 +176,18 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 	// verify nodes count restoration even if machine == nil.
 	if machine == nil || machine.GetCreationTimestamp().After(mdr.GetCreationTimestamp().Time) {
 		if isRestored, err := r.isExpectedNodesNumberRestored(ctx, mdr); err != nil {
-			log.Error(err, "could not verify if node was restored")
+			msg := "could not verify if node was restored"
+			log.Error(err, msg)
+			commonevents.WarningEvent(r.Recorder, mdr, "unableToVerifyNodesCount", err.Error())
 			if err == unrecoverableError {
 				return ctrl.Result{}, nil
 			}
 			return ctrl.Result{}, err
 		} else if isRestored {
 			_, err = r.updateConditions(remediationFinishedMachineDeleted, mdr)
+			if err == nil {
```

Comment:
NIT: I am not sure it is needed. 
Function `updateConditions` returns an error only when it was called with unknown conditionChangeReason when `remediationFinishedMachineDeleted` reason is already found/defined in the function.